pydantic/config.py

```diff
@@ -709,11 +709,30 @@ class Model(BaseModel):
 
     defer_build: bool
     """
-    Whether to defer model validator and serializer construction until the first model validation.
+    Whether to defer model validator and serializer construction until the first model validation. Defaults to False.
 
     This can be useful to avoid the overhead of building models which are only
     used nested within other models, or when you want to manually define type namespace via
-    [`Model.model_rebuild(_types_namespace=...)`][pydantic.BaseModel.model_rebuild]. Defaults to False.
+    [`Model.model_rebuild(_types_namespace=...)`][pydantic.BaseModel.model_rebuild].
+
+    See also [`defer_build_mode`][pydantic.config.ConfigDict.defer_build_mode].
+
+    !!! note
+        `defer_build` does not work by default with FastAPI Pydantic models. Meaning the validator and serializer
+        is constructed immediately when the model is used in FastAPI routes instead of construction being deferred.
+        You also need to use [`defer_build_mode='always'`][pydantic.config.ConfigDict.defer_build_mode] with
+        FastAPI models.
```

Comment:
I think the mention of FastAPI is helpful here, but we should probably explain that this is because of FastAPI's use of `TypeAdapter`s

Reply:
Done